tck/src/main/java/org/eclipse/configjsr/CDIPropertyNameMatching.java

```diff
@@ -0,0 +1,127 @@
+ * among environment variables, the variable name needs to be the same with case sensitive (e.g. com_ACME_size). If not found, try to match 
+ * the case-insensitive environment variable (e.g. COM_ACME_SIZE or COME_ACME_Size)
+ * 
+ * Prerequisite: 
+ * The following environment variables must be set prior to running this test:
+ * "my_int_property" with the value of "45"
+ * "MY_BOOLEAN_Property" with the value of "true"
+ * "my_string_property" with the value of "haha"
+ * "MY_STRING_PROPERTY" with the value of "woohoo"
+ * @author Emily Jiang
+ */
+public class CDIPropertyNameMatching extends Arquillian {
+
+
+    @Deployment
+    public static Archive deployment() {
+        return ShrinkWrap.create(WebArchive.class)
+                .addClasses(CDIPropertyNameMatching.class, SimpleValuesBean.class)
+                .addAsManifestResource(new StringAsset(
+                        "my.int.property=3"+
+                        "\nmy.string.property=fake" +
+                        "\nmy.random.string.property=random"),
+                            "microprofile-config.properties")
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+    }
+    
+    @BeforeClass
+    public void checkSetup() {
+       //check whether the environment variables were populated by the executor correctly
+    
+        if (!"45".equals(System.getenv("my_int_property"))) {
+         Assert.fail("Before running this test, the environment variable \"my_int_property\" must be set with the value of 45");
+        }
+        if (!"true".equals(System.getenv("MY_BOOLEAN_Property"))) {
+            Assert.fail("Before running this test, the environment variable \"MY_BOOLEAN_Property\" must be set with the value of true");
+        }
+        if (!"haha".equals(System.getenv("my_string_property"))) {
+            Assert.fail("Before running this test, the environment variable \"my_string_property\" must be set with the value of haha");
+        }
+        if (!"woohoo".equals(System.getenv("MY_STRING_PROPERTY"))) {
+            Assert.fail("Before running this test, the environment variable \"MY_STRING_PROPERTY\" must be set with the value of woohoo");
+        }
+
+    }
+
+    @Test
+    public void can_inject_simple_values_when_defined() {
+        SimpleValuesBean bean = getBeanOfType(SimpleValuesBean.class);
+
+        assertThat(bean.stringProperty, is(equalTo("haha")));
+        assertThat(bean.booleanProperty, is(true));
+        assertThat(bean.intProperty, is(equalTo(45)));
+        assertThat(bean.randomStringProperty, is(equalTo("random")));
+    }
+
+
+
+    private <T> T getBeanOfType(Class<T> beanClass) {
+        return CDI.current().select(beanClass).get();
+    }
+
+    @Dependent
+    public static class SimpleValuesBean {
+
+        @Inject
+        @ConfigProperty(name="my.string.property")
+        private Provider<String> stringProperty;
```

Comment:
now that env variable must be set by the test runner, do you see another reason to use `Provider` instead of plain fields?